### ISSUE TITLE
perf(mesh): store interior/safe_neighbour on FaceCellConnectivity and drop redundant scatter masks

### ISSUE THREAD

**Issue by ctdegroot** [DeGrootResearchGroup/aquaflux#109]
Surfaced by a code-wide speed review.

## 1. `interior` / `safe_neighbour` are recomputed properties

`aquaflux/mesh/connectivity.py:110`:

```python
@property
def safe_neighbour(self):
    return jnp.where(self.interior, self.neighbour, self.owner)
```

`safe_neighbour` has 19 call sites, and each access recomputes `neighbour >= 0` plus a select over `(n_faces,)`. Outside a loop body XLA folds this away, so there is no cost there. **Inside** the GMRES/matvec `while_loop` bodies of `CorrectedGreenGauss.operator` and `HessianCorrectedGradient` it does not fold: the compare and the select run per iteration.

Both `interior` and `safe_neighbour` are pure functions of `owner`/`neighbour`, which are fixed at mesh build time. Storing them as fields computed once in the constructor makes them enter loop bodies as parameters rather than recomputed values.

The docstring at :108 already promises this behaviour — "Gather several fields off it without recomputing the substitution each time" — and the property does not deliver it.

## 2. `scatter` masks with a full-array `where`

`aquaflux/mesh/connectivity.py:186-190`:

```python
mask = _broadcast_face_mask(self.interior, neighbour_contrib.ndim)
neigh = jnp.where(mask, neighbour_contrib, 0.0)
return segment_sum(owner_contrib, self.owner, self.n_cells) + segment_sum(
    neigh, self.safe_neighbour, self.n_cells)
```

The `where` can be eliminated entirely: precompute a `neighbour_scatter_index` that maps boundary faces to row `n_cells`, scatter with `num_segments=n_cells+1`, and slice `[:n_cells]`. `segment_sum` then discards the boundary rows structurally, with no select and no extra `(n_faces, ...)` allocation. Every residual term does at least one scatter.

## 3. Two wasted full-face selects in the limiter

`aquaflux/schemes/limiter.py:109,114`:

```python
return jnp.where(active, numerator / denominator, jnp.inf)          # :109
...
psi = face_cells.scatter_min(
    face_limiter(owner, jnp.ones_like(interior)), face_limiter(neighbour, interior))  # :114
```

- `jnp.ones_like(interior)` materializes an all-`True` `(n_faces,)` bool array purely to drive a select whose answer is always the first branch. The owner side is *always* active and needs no mask at all.
- The neighbour side is masked to `+inf` **twice**: once by `face_limiter`'s `where(active, ..., inf)`, then again by `_scatter_extremum` (`connectivity.py:267`), which does its own `jnp.where(mask, neighbour_contrib, identity)` with the same `+inf` identity.

The limiter runs on every residual evaluation whenever `LimitedUpwind` is the active advection scheme, so this is on the true hot path. Impact is bounded (three elementwise face-width ops) but the fix is free: make `active` optional (`None` => no mask) and drop the redundant one.

## Acceptance

- `interior` and `safe_neighbour` are stored fields; no recomputation appears inside the gradient-solve `while_loop` bodies in optimized HLO.
- `scatter` performs no masking select.
- Bit-identical results across the existing mesh, scheme and limiter unit tests.
